hardware/function_generators.py

```diff
@@ -18,9 +19,108 @@
 
 import visa
 import numpy as np
+import random
 
+from hardware import u
 
-class Agilent_33250A():
+class MockFunctionGenerator:
+
+    def __init__(self, instr_name = None):
+        if not instr_name:
+            self.name = "AWG - Agilent_33250A"
+        else:
+            self.name = instr_name
+        self._frequency = 3 * u.hertz
+        self._voltage = 1 *u.volt
+        self.waveforms = ['SIN', 'SQU', 'RAMP', 'PULS', 'NOIS', 'DC', 'USER']
+        self._phase = 20 * u.degree
+        self._duty_cycle = 50 #as a percent
+
+    def identify(self):
+        return self.name
+
+    @property
+    def frequency(self):
+        return self._frequency
+
+    @frequency.setter
+    def frequency(self, val):
+        self._frequency = val * u.hertz
+
+    @property
+    def voltage(self):
+        return self._voltage
+
+    @voltage.setter
+    def voltage(self, val):
+        self._voltage = val *u.volt

```

Comment:
add a space after the *

Reply:
Maybe by installing a linter...

Reply:
yea, I'd look into [flake8](http://flake8.pycqa.org/en/latest/)